CDP4DiagramEditor/ViewModels/DiagramItemDataTemplateSelector.cs

```diff
@@ -43,22 +60,13 @@ public class DiagramItemDataTemplateSelector : DataTemplateSelector
         /// <returns>The <see cref="DataTemplate"/> to use</returns>
         public override DataTemplate SelectTemplate(object item, DependencyObject container)
         {
-            if (!(item is NamedThingDiagramContentItem))
+            return item switch
             {
-                return base.SelectTemplate(item, container);
-            }
-            
-            return item is DiagramPortViewModel ? this.DiagramPortTemplate : this.GenericDiagramItemDataTemplate;
+                ElementDefinitionDiagramContentItem => this.ElementBaseDiagramItemTemplate,
```

Comment:
In this situation its more usual to create a template in xaml per type using the template DataType parameter. Not sure if that's possible here

Reply:
DevExpress states that using a TemplateSelector is the way to go.